packages/lisk-p2p/src/p2p_types.ts

```diff
@@ -105,6 +105,25 @@ export interface ProtocolNodeInfo {
 	readonly options?: P2PInfoOptions;
 }
 
+export interface P2PPeerSelectionOptions {
+	readonly [key: string]: string | number;
```

Comment:
Should the value be restricted to only string or number? Maybe it should be `unknown`? Or maybe the `selectionParams` itself should be `unknown` instead of `P2PPeerSelectionOptions`?

Or if unknown doesn't work, maybe we can make an exception to TSLint here and allow `any`? Then it would be the user's responsibility to enforce type constraints.

The user should be able to override our types with a more specific (but compatible) type in their function declaration.

Reply:
Instead of using unknown or any, we can also add `object` along with `string` and `number`?

Reply:
As discussed, consider making the selection function accept `P2PNodeInfo` as a parameter instead of `P2PPeerSelectionOptions/PeerOptions`. Then the selection function should use `nodeInfo.height` instead of `selectionParams.lastBlockHeight`. That would allow the selection function to operate on any custom field which might be in the nodeInfo's `options` object.